### PR TITLE
spray now checks if the guy is injectable

### LINKED ISSUE #46275: Spray Bottles (and similar items) ignore any protections you have and inject chems
Not sure if this is a bug or intentional, but there is a real possibility to simply inject various chems into people with a spray bottle regardless of whatever protective measures you attempt to take and whether you have internals on or not. This bypasses even hardsuits sealed up with internals on.

We have a large variety of applicators that don't do this, so I'm pretty sure for the sake of consistency this seems to stand out rather strongly, especially given spray bottles are widely available in the service lathe.

This includes stuff like poisons (if it sprays enough), holy water, frost oil, mutation toxin (this is a hard stun by the way), cobbychems that hurties your organs, honey, nutrients, mint toxin, gluttony's blessing, baccus (it can set you on fire since it's touch applied too so look out for CLF3), advanced slime mutation toxin, frost oil (hardsuits are especially at risk for this).

Loads of nasty things honestly and this probably includes anything that is a subclass of sprayers.

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

you cant spray through thick clothing and stuff like that anymore
fixes #46275

## Why It's Good For The Game

hmm today i will spray a guy with a hardsuit

## Changelog
:cl:
fix: you cant spray through thick clothing anymore
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
now make it apply to patches

**Comment by ExcessiveUseOfCobblestone** [comment]
> now make it apply to patches

based, redpatched

File: code/modules/reagents/reagent_containers/spray.dm [hunk 1]
```diff
@@ -90,6 +90,8 @@
 			if(stream_mode)
 				if(isliving(T))
 					var/mob/living/M = T
+					if(!M.can_inject())
+						return
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/reagent_containers/spray.dm]: this should prob be `continue` instead

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/reagent_containers/spray.dm]: continue means continue to the next item in the loop, not continue to the next block of code below

Comment (Fikou) [reply code/modules/reagents/reagent_containers/spray.dm]: ok i guess

**Comment by ExcessiveUseOfCobblestone** [comment]
if you tested it you wouldn't have to guess 😏

**Comment by adamkad1** [comment]
i though patches used the power of bluespace bullshit (and a lengthy delay)

**Pull request merged**